pkg/devfile/validate/generic/components.go

```diff
@@ -0,0 +1,78 @@
+package generic
+
+import (
+	"fmt"
+	"strings"
+
+	adaptersCommon "github.com/openshift/odo/pkg/devfile/adapters/common"
+	"github.com/openshift/odo/pkg/devfile/parser/data/common"
+	"github.com/openshift/odo/pkg/util"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
```

Comment:
we should not import odo packages in library, move this function in generic package `util.ValidateK8sResourceName`

```
"github.com/openshift/odo/pkg/util"
```

not sure about this one, anyways the constants are not related with adapters, it is a devfile spec so can be moved in types package.
```
"github.com/openshift/odo/pkg/devfile/adapters/common
```

for below one i understand, it will get updated when we use types from devfile/api
```
github.com/openshift/odo/pkg/devfile/parser/data/common
```

Reply:
1. So devfile/parser already has a copy of the util func ValidateK8sResourceName https://github.com/devfile/parser/blob/master/pkg/util/util.go#L1044. I plan on using that when its exported out. It will just be changing the import.
2. `adaptersCommon "github.com/openshift/odo/pkg/devfile/adapters/common"` are just using constants. There will be new const when its moved to devfile/parser. Since this is still in odo, I dont want to maintain two separate const for the same thing.

Lmk, what you think @adisky

Reply:
gotcha!